### ISSUE TITLE
Autoscroll during drag-and-drop of a clause

### ISSUE THREAD

**Issue by dselman** [accordproject/web-components#133]
# Feature Request 🛍️
Drag-and-drop of clauses within a contract is of limited value, because the contract doesn't autoscroll when you get near the top or the bottom of the visible part of the contract.

## Use Case
Create a long contract. Decide you'd like to move a clause from the beginning to the end of the contract.

## Possible Solution
Some sort of autoscroll of the contract.

**Comment by sanketshevkar** [comment]
Hello @dselman I would like to work on this feature. Is this still open?

**Comment by jeromesimeon** [comment]
> Hello @dselman I would like to work on this feature. Is this still open?

Hi @sanketshevkar As far as I can tell this is still open. You can test it on the storybook.

https://ap-web-components.netlify.app/?path=/story/contract-editor--contract-editor

I'll assign you. CC @irmerk

**Comment by sanketshevkar** [comment]
Ok cool. Thanks @jeromesimeon

**Comment by TC5022** [comment]
Hi @dselman, I would like to work on this issue if it is still open

**Comment by Michael-Grover** [comment]
@sanketshevkar are you still working on this issue? If not, I will assign to @TC5022

**Comment by TC5022** [comment]
@sanketshevkar are you working on it?

**Comment by sanketshevkar** [comment]
Sorry @TC5022 @Michael-Grover. Nope I'm not working on this issue.

**Comment by TC5022** [comment]
@irmerk @Michael-Grover thank you for assigning me the issue. Is it possible to contact you regarding some doubts related to this?

**Comment by Michael-Grover** [comment]
Hi @TC5022 , if you have questions, you can put them in this issue

**Comment by TC5022** [comment]
@Michael-Grover @irmerk I think the issue could be solved using the [autoscroll-react ](https://www.npmjs.com/package/autoscroll-react) package but according to their documentation, it uses a ref. So, I am thinking to add the useRef hook. Does it sound good to proceed?

**Comment by TC5022** [comment]
Also, I am facing a little trouble trying to figure out which file to edit. Would you be able to guide me regarding that? @Michael-Grover @irmerk

**Comment by Michael-Grover** [comment]
@dselman @DianaLease CC'in you.

**Comment by TC5022** [comment]
The changes I am making in the contract editor code aren't reflected in the storybook. Could you please guide me as to why this is happening? @dselman @DianaLease

**Comment by dselman** [comment]
Did you check the README? I think there are some notes there for what you need to do to rebuild.

**Comment by TC5022** [comment]
Yes i resolved it, there was an issue with my system. Sorry for the trouble.

**Comment by irmerk** [comment]
Please open a PR when you have something with `autoscroll-react` so we can review how that could work here.

**Comment by TC5022** [comment]
Sure, I have almost worked up the code. Will open a PR as soon as I get it finalised

**Comment by KhushV25** [comment]
Hey Can I work on this or is this solved?
